### PR TITLE
Add -P option to nosetests

### PR DESCRIPTION
**Pull request by wizmer**
Add \_\_init\_\_.py in tests folder

Also:
use pathlib instead of pathlib2

Kudos to Mike for the discover magic.

### CODE REVIEW

**Comment by wizmer** [comment]
I have no idea of what I'm doing.

**Comment by mgeplf** [comment]
What was the initial problem that this was trying to solve?  'Tests can run from any location' was the name of my PR in libsonata, b/c the `def test_suite` was hardcoding a path.

**Comment by wizmer** [comment]
I had some issue in the CI wheel PR because the tests.utils module was not found so I had to add an \_\_init.py\_\_ in tests/ which in turn broke `ci/test_python.sh`.
The discover magic allows to both have \_\_init.py\_\_ in the tests folder and `ci/test_python.sh` working.

**Comment by asanin-epfl** [comment]
In our Python projects we stay within the paradigm of `from utils import` not `from . utils import`. Is `.` usage necessary? Anyway, @mgeplf should say the final word here. I don't quite understand what's happening.

**Comment by wizmer** [comment]
For some reason that I did not understand, the tests, when run as part of the Windows wheel building are not passing with the old way.
https://github.com/BlueBrain/MorphIO/runs/1493403378?check_suite_focus=true#step:9:697

I end-up with a lot of:
```
File "D:\a\MorphIO\MorphIO\tests\test_7_modifiers.py", line 8, in <module>
  from utils import captured_output
ModuleNotFoundError: No module named 'utils'
```
So I tried a bunch of stuff and this worked...

**Comment by asanin-epfl** [comment]
How urgent is this?

**Comment by wizmer** [comment]
https://github.com/BlueBrain/MorphIO/pull/214 will need to be rebased on top of this commit.
https://bbpteam.epfl.ch/project/issues/browse/NSETM-1018 is now on priority high so I guess somewhat urgent.

**Review by asanin-epfl — APPROVED** [review]

File: tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import unittest
+
+
+def test_suite():
+    loader = unittest.TestLoader()
+    suite = loader.discover('tests', pattern='test*.py')
```
Comment (mgeplf) [inline tests/__init__.py:6]: isn't this hardcoding the directory?

**Comment by mgeplf** [comment]
Looking at this more; I think the fix is the `-P` and the correct relative imports.  Can you test this? It identifies the root cause much more clearly.  Then, we need to figure out what they -P is needed.

**Comment by wizmer** [comment]
I've just tested this. Without the `-P`, nosetests append the `src` folder at the beginning of $PATH. Since there is a morphio folder in src, it will find this one first instead of finding the true morphio package installed in site-packages.

**Comment by mgeplf** [comment]
That seems a lot cleaner, thanks.

**Pull request merged**